api/lcp/server.py

```diff
@@ -0,0 +1,351 @@
+                protected_content_sha256=encrypted_content.protected_content_sha256
+            )
+            path = '/contents/{0}'.format(encrypted_content.content_id)
+
+            self._send_request(configuration, 'put', path, payload, LCPEncryptorResultJSONEncoder)
+
+    def generate_license(self, db, content_id, patron_id, license_start, license_end):
+        """Generates a new LCP license
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param content_id: Unique content ID
+        :type content_id: string
+
+        :param patron_id: Unique patron ID
+        :type patron_id: string
+
+        :param license_start: Unique patron ID
+        :type license_start: string
+
+        :param license_start: Date and time when the license begins
+        :type license_start: datetime.datetime
+
+        :param license_end: Date and time when the license ends
+        :type license_end: datetime.datetime
+
+        :return: LCP license
+        :rtype: Dict
+        """
+        with self._configuration_factory.create(
+                self._configuration_storage, db, LCPServerConfiguration) as configuration:
+            partial_license_payload = self._create_partial_license(configuration, patron_id, license_start, license_end)
+            path = 'contents/{0}/license'.format(content_id)
+            response = self._send_request(configuration, 'post', path, partial_license_payload)
+
+            return response.json()
+
+    def get_license(self, db, license_id):
+        """Returns an existing license
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param license_id: License's ID
+        :type license_id: int
+
+        :return: Existing license
+        :rtype: string
+        """
+        with self._configuration_factory.create(
+                self._configuration_storage, db, LCPServerConfiguration) as configuration:
+            partial_license_payload = self._create_partial_license(configuration)
+            path = 'licenses/{0}'.format(license_id)
+
+            response = self._send_request(configuration, 'post', path, partial_license_payload)
+
+            return response.json()
+
+    def get_licenses(self, db):
+        """
+        Returns existing licenses
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: List of existing licenses
+        :rtype: List[string]
+        """
+        with self._configuration_factory.create(
+                self._configuration_storage, db, LCPServerConfiguration) as configuration:
+            page_index = 1
+            licenses = []
+
+            while True:
+                path = 'licenses?page={0}&per_page={1}'.format(page_index, configuration.lcpserver_page_size)
+                response = self._send_request(configuration, 'get', path, payload=None)
+                new_licenses = response.json()
```

Comment:
This may not be under your control, but how does the pagination work here? If licenses are added to the list while you're iterating, you will miss some. Is this a problem? I don't see `get_licenses` used elsewhere in this code.

Reply:
That's a very good question. So far `get_licenses` is not used, I was planning to use in `LCPAPI.patron_activity` but ended up with querying loans from the database

Reply:
OK, take out the code and its tests since we're not using it.

Reply:
Removed the code